Author

Berke94 opened the pull request:
2. aşama dan 
-Yazarlar ayrı tabloda tutuluyor (one to many)
-Kitap ekleme ve düzenleme sayfalarında yazarın adının bir kısmı yazılınca DB 'den alıyor
-Yazar onay sayfası var(ancak işlevsel çalışmıyor sadece onay tıklıyorsun DB de 0 dan 1 dönüyor)
-Satıldığı kitapçılar eklendi pilot tablo ile yaptım (3 Adet ekledim senin yazdıklarını orayı seeder olarak ekledim doğru bir mantık mı tam emin değilim ama 3 kitapçı için çalışıyor) (many to many)
-Düzenleme ekranında da mevcut tekrar düzenlerken ekleniyor kitapçılar

Berke94 committed:
edit controller

Berke94 committed:
add is_active and seeder

Berke94 committed:
add is_active and seeder

Berke94 committed:
command

Berke94 committed:
command

Berke94 committed:
Breeze Eklendi - Login ve register sayfaları düzenlendi - route lar ayarlandı

Berke94 committed:
middleware(login olmadan books ulaşılamıyor) - Ana sayfa ekleme ve ufak tefek düzenleme - Logo ekleme

Berke94 committed:
Yazarlar ayrı bir tabloya alındı
Bir yazar için authors tablosunda tek yer tutuluyor ve books tablosuyla ilişkilendirildi

Berke94 committed:
Navbar ayrı yere alındı
scriptler ayrı yere alındı
Yazar onay sayfası eklendi

Berke94 committed:
many to many pivot tablo ile kitapçılar yapıldı
3 adet kitapçı listesi seeder olarak eklendi
seçilebiliyor 3 de listelenebilyior
düzenlemede eklenebiliyor

Berke94 committed:
Sayfalar düzenlendi
Butonlar değiştirildi ve iconlar eklendi
Kullanıcı girişi yapılmadan Kitaplar listesine ve Yazar onayla sayfasına girilme ayarlandı
Sayfalar biraz daha modern hale getirildi.

Berke94 committed:
yazar onay sayfasında Uyarı renklendirildi silinince kırmızı onaylayınca yeşil

Berke94 committed:
Kitapları Yazar Adına - Eklenme Tarihine - Alfabetik sıraya göre Sıralama ekledim.

semihkeskindev commented on app/Http/Controllers/AuthorController.php:
```diff
@@ -0,0 +1,64 @@
+    public function store(Request $request)
+    {
+        $request->validate([
+            'name' => 'required|string|max:255|unique:authors,name',
```
kuralları | ile tek string olacak şekilde yazmak yerine ['required', 'string'...] olacak şekilde array içerisinde ayrı ayrı yazman daha iyi olacaktır kuralları.

semihkeskindev replied:
+ form request sınıflarını kullanalım validationlarda https://laravel.com/docs/11.x/validation#creating-form-requests

Berke94 replied:
O tarafı sildiğimiz için bir değişiklik ekleyemedim ama anladığım şu şekil kullanım yap diyorsun ?
'name' => ['required', 'string', 'max:255', 'unique:authors,name']
Döküman tarafında şu biraz kafamı karıştırdı hiç kullanmadığım için hiç Requests tarafına girişmiş

Berke94 replied:
BookController tarafınıda düzenledim bu şekilde

semihkeskindev commented on app/Http/Controllers/AuthorController.php:
```diff
@@ -0,0 +1,64 @@
+            'name' => $request->input('name'),
+        ]);
+
+        return response()->json($author); // JSON yanıt döndür
```
rest api yazmıyoruz niye json response döndürüyorsun? önceki sayfaya geri yönlendirelim form submit edilince sayfaya geri dönsün böylece

Berke94 replied:
Bu kısma taha ile baktık store tarafı zaten buradan kullanılmıyor komple kaldırdık. search tarafını return $results; olarak değiştirdik json la bir iş yokmuş aslında biraz benim GPT tarafından sarpa sarmam olmuş.

semihkeskindev commented on app/Http/Controllers/AuthorController.php:
```diff
@@ -0,0 +1,64 @@
+
+        return response()->json($author); // JSON yanıt döndür
+    }
+    public function search(Request $request)
```
Bu rest api endpointlerini, Controllers altında "Api" klasörü oluşturup içine yeni bir XYZController ekleyerek (örneğin AuthController) onda tut. Rest Api ile web endpointleri Controllerlarda karışmasın.

semihkeskindev commented on app/Http/Controllers/AuthorController.php:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    // Yazar onaylama işlemi
+    public function approve($id)
```
Route model binding kullan https://laravel.com/docs/11.x/routing#route-model-binding

Berke94 replied:
Bunu GPT desteksiz kendim hallettim. Ama bazı kaynaklarda Route Model Binding kullanmanın kontrolü kısıtladığı için tercih edilmediği yazıyordu. Bir ara anlatırsan iyi olabilir durumu neden öyle denildiğini ya da neden denilen şeyin yanlış olduğunu - destroy tarafınada uyguladım

semihkeskindev commented on app/Http/Controllers/AuthorController.php:
```diff
@@ -0,0 +1,64 @@
+        $author = Author::findOrFail($id);
+        $author->delete();
+
+        return redirect()->route('authors.approval')->with('danger', 'Yazar başarıyla silindi.');
```
burada niye danger kullandın? yazarı siliyor olmamız durumu olumsuz yapmıyor. yine success olmalı sonuç olarak başarıyla silinmiş oluyor. olumlu bir durum.

Berke94 replied:
Success kullanmıştım ilk başta ama sonra kırmızı yapmak istedim açılan bar kısmını nasıl yapılacağına bakarken danger olarak çıktı karşıma

semihkeskindev replied:
düzeltirsin onu o zaman

semihkeskindev commented on app/Http/Controllers/BookController.php:
```diff
@@ -0,0 +1,128 @@
+            'isbn_number' => 'required|string|max:255',
+            'number_of_pages' => 'required|integer',
+            'bookstores' => 'nullable|array',
+            'bookstores.*' => 'exists:bookstores,id'
```
exists kuralının öncesinde "integer" olduğunun kontrolünü yaparsan db ye sorgu atmadan önce verinin valid formatta olduğu doğrulamış oluruz gereksiz veritabanını yormaktan kaçınırız.

semihkeskindev commented on app/Http/Controllers/ProfileController.php:
```diff
@@ -0,0 +1,61 @@
+     */
+    public function destroy(Request $request): RedirectResponse
+    {
+        $request->validateWithBag('userDeletion', [
```
bu method ne işe yarıyor bu mesajı görünce bana detaylıca anlatabilir misin ben bilmiyorum öğretmeni istiyorum :) @Berke94

Berke94 replied:
ProfileController dosyası breeze ile birlikte geldi üzerinde bir oynamayı ben yapmadım. Devamına baktığımda şifre girilip doğrulama yapınca kullanıcı siliniyor.

semihkeskindev replied:
olabilir o methodu kendi controller sınıflarımızda da kullanabiliriz. senin yapmana gerek yok. sadece inceleyip/araştırıp anlatmanı istiyorum telegram üzerinden ne işe yarıyor. ->validate methodundan farkı nedir?

semihkeskindev commented on database/migrations/2024_07_24_104411_create_books_table.php:
```diff
@@ -13,8 +13,8 @@ public function up(): void
-            $table->string('book');
-            $table->string('author_name');
+            $table->string('book_name');
+            $table->string('author');
```
- book_name isinmlendirmesi pratik değil. Sebebi ise zaten books tablosundaki kolonlar aksi belirtilmedikçe book ile alakalı datalardır. nasıl id nin başına extra book_id diye yazılmasına gerek yoksa zaten kitapların id si olduğu anlaşılııyorsa name için de aynısı geçerli. 

İstisna durumları author_name gibi bilgiler tutacağın zaman olur.

(Değişiklik yapmana gerek yok sadece bil yeter)

Berke94 committed:
-Store buradan çalışmadığı için kaldırdık (Taha ile birlikte baktık ve seearch tarafında return $results; a çevirdik

Berke94 committed:
-Danger benim yanlış kullanımımdı onu tekrardan success olarak düzelttim

Berke94 committed:
exists kuralının öncesinde "integer" olduğunun kontrolünü eklendi
anladığım şekilde ve gönderdiğin dökümana bakarak array içerisinde ayrı ayrı toplandı.

Berke94 committed:
Route model binding olarak düzenlendi
